src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -55,38 +55,47 @@ Whenever comparing tickets is evoked in Filecoin, for instance when discussing s
-- While true:
-    - Set referenceTipsetHeight = n - referenceTipsetOffset
-    - If blocks were mined at referenceTipsetHeight:
-        - ReferenceTipset = TipsetAtHeight(referenceTipsetHeight)
-        - Select the block in ReferenceTipset with the smallest final ticket, return its ticket (pastTicket).
-    - If no blocks were mined at referenceTipsetHeight:
-        - Increment referenceTipsetOffset
-        - (Repeat)
-- newRandomness = H(pastTicket || minerAddress || n)
+```text
+Set referenceTipsetOffset = 0
+While true:
+    Set referenceTipsetHeight = n - referenceTipsetOffset
+    If blocks were mined at referenceTipsetHeight:
+        ReferenceTipset = TipsetAtHeight(referenceTipsetHeight)
+        Select the block in ReferenceTipset with the smallest final ticket, return its value (pastTicket).
+    If no blocks were mined at referenceTipsetHeight:
+        Increment referenceTipsetOffset
+        (Repeat)
+newRandomness = H(pastTicket || n)
+```
 
 In english, this means two things:
+- Choose the smallest ticket in the Tipset if it contains multiple blocks.
 - When sampling a ticket from an epoch with no blocks, draw the min ticket from the prior epoch with blocks and concatenate it with
-    - the minerAddress of the miner using this input
     - the wanted epoch number
     - hash this concatenation for a usable ticket value
-- Choose the smallest ticket in the Tipset if it contains multiple blocks.
 
 See the `RandomnessAtEpoch` method below:
 {{< readfile file="block.go" code="true" lang="go" >}}
 
-The above means that ticket randomness is reseeded with every new block, but can indeed be derived by any miner for an arbitrary epoch number using a past epoch. However, this does not affect protocol security under Filecoin's clock synchrony assumption.
+The above means that ticket randomness is reseeded with every new block, but can indeed be derived by any miner for an arbitrary epoch number using a past epoch.
 
 {{<label ticket_generation>}}
-### Ticket generation
+### Randomness Ticket generation
 
 This section discusses how tickets are generated by EC for the `Ticket` field in every block header.
 
 At round `N`, a new ticket is generated using tickets drawn from the Tipset at round `N-1` (as shown below).
 
-The miner runs the prior ticket through a Verifiable Random Function (VRF) to get a new unique ticket which can later be derived for randomness (as shown above).
+The miner runs the prior ticket through a Verifiable Random Function (VRF) to get a new unique ticket which can later be derived for randomness (as shown above). The prior ticket is prepended with the ticket domain separation tag `1` and concatenated with the miner actor address (to ensure miners using the same worker keys get different randomness).
+
+To generate a ticket for a given epoch n:
+```text
+LastTicket = MinTicketValueAtEpoch(n-1)
+newRandomness = VRF_miner(H(TicketDST || Serialization(pastTicket, minerActorAddress)))
```

Comment:
(as above, ideally would be specified in terms of `DeriveRand`)

Reply:
It is in the code. 

Again I don't think we should force people to your abstractions. The prose should also standalone.

Reply:
Up to you, I think it's clearer to the reader if we have a single `DeriveRand` primitive in the prose, but I don't have a strong preference as long as it's functionally correct (see https://github.com/filecoin-project/specs/pull/677#discussion_r352750763).